accepted/2022/non-root-container-images.md

```diff
@@ -0,0 +1,162 @@
+# Proposal for .NET Non-Root Container Images
+
+We all know that [running as root is considered harmful](https://unix.stackexchange.com/questions/1052/concern-about-logging-in-as-root-overrated) yet it is commonplace to [run container images that way](https://www.redhat.com/blog/understanding-root-inside-and-outside-container). We can make it easier to run .NET container images as non-root.
+
+## Goals
+
+The following will improve security for users:
+
+- Enable all .NET images to be run as a non-root user.
+- Exposed ports (via `ASPNETCORE_URLS`) are constant across all image types.
+- Images do not use privileged resources by default.
+- It is easy to use any combination of root and non-root images in a deployment.
+
+## Plan
+
+We can satisfy those goals (not necessarily in that order) across multiple releases.
+
+- .NET 7
+  - Add same non-root user in both rootful and non-root images
+  - Expose same non-root ports -- `8080` and `8443` (for HTTPS) -- in both rootful and non-root images
+  - Continue to expose port `80` in rootful images, for compatibility.
+  - Transition `mcr.microsoft.com/dotnet/samples` to non-root images (which by definition means not exposing port `80`).
+- .NET 8
+  - Remove port `80` from rootful images.
+  - Announce that all .NET images are turnkey non-root capable.
```

Comment:
@richlander - Since none of these changes happened for .NET 7, what's the plan for .NET 8. Do we want to continue a phased approach between .NET 8 and 9 or do everything in .NET 8?

Reply:
If we make these changes for Preview 1 or 2, then I think we can do everything at once and wait for feedback and then assess based on that.

Reply:
cc @lbussell